src/openrct2/core/BitSet.hpp

```diff
@@ -0,0 +1,545 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2021 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <limits>
+#include <string>
+
+namespace OpenRCT2
+{
+    namespace Detail
+    {
+        namespace BitSet
+        {
+            static constexpr size_t byte_bits = std::numeric_limits<std::underlying_type_t<std::byte>>::digits;
```

Comment:
i know we said that nothing at present conforms to much of a style for these but pretty much nothing should be snake_case.

Reply:
I wrote this code in STL style, I can change it all.

Reply:
I renamed everything to the point of not breaking STL compatibility